### ISSUE TITLE
App stops screensaver from activiating on MacOS

### ISSUE THREAD

**Issue by palmerj** [joncampbell123/dosbox-x#4243]
### Describe the bug

On MacOS the app stops the OS from running it's screensaver.  Please implement something like this in dosbox-x to remove this limitation https://github.com/dosbox-staging/dosbox-staging/pull/717/files

Workaround for now is to run dosbox-x with `SDL_VIDEO_ALLOW_SCREENSAVER=1 /Applications/dosbox-x.app/Contents/MacOS/dosbox-x &`

### Steps to reproduce the behaviour

1. Ensure screensaver is set to run in MacOS settings
2. Run dosbox-x
3. Check activity monitor and check dosbox-x process in energy tab. It reports Preventing Sleep as `Yes`

### Expected behavior

Screensaver should work when running dosbox-x.

### What operating system(s) this bug have occurred on?

MacOS 13.1

### What version(s) of DOSBox-X have this bug?

DOSBox-X 2023.03.31

### Have you checked that no similar bug report(s) exist?

- [X] I have searched and didn't find any similar bug report.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by joncampbell123** [comment]
Good idea. Knowing the SDL library it made sense for them to do that, but yes, perhaps let's give the option to allow screensaver.
